src/analysis/retail/warrior/arms/modules/features/RageTracker.tsx

```diff
@@ -0,0 +1,85 @@
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/warrior';
+import HIT_TYPES from 'game/HIT_TYPES';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent, DamageEvent } from 'parser/core/Events';
+import ResourceTracker from 'parser/shared/modules/resources/resourcetracker/ResourceTracker';
+
+const RAGE_PER_MELEE_HIT = 25;
+
+class RageUsage extends ResourceTracker {
+  lastMeleeTimestamp = 0;
+
+  private usesWarlordsTorment = false;
+  private usesWarMachine = false;
+
+  constructor(options: Options) {
+    super(options);
+    this.resource = RESOURCE_TYPES.RAGE;
+    this.usesWarlordsTorment = this.selectedCombatant.hasTalent(TALENTS.WARLORDS_TORMENT_TALENT);
+    this.usesWarMachine = this.selectedCombatant.hasTalent(TALENTS.WAR_MACHINE_ARMS_TALENT);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.MELEE), this.onDamage);
```

Comment:
this should probably be named `onAutoAttackDamage` just for clarity because it won't be correct if you added it to another damage event listener